examples/forms.rs

```diff
@@ -0,0 +1,225 @@
+
+        // because this is a terminal field, we must finish it as an annotation
+        // to provide it's location, size and optionally appearance, most
+        // readers highlight form fields, so the appearance is optional, the
+        // created annotation type is already set to widget, so we don't write
+        // it ourselves
+        let mut annot = field.to_annotation();
+        annot.rect(Rect::new(108.0, 730.0, 208.0, 748.0));
+
+        // TODO: test whether this is actually needed depending on the
+        // appearance stream
+
+        // we ensure that the annoation is printed too
+        annot.flags(AnnotationFlags::PRINT);
+    }
+
+    // a good form has radio buttons, radio button fields are related by their
+    // common parent, each set of buttons needs a unique parent
+    let radio_id = Ref::new(5);
+    let radio_ids = [Ref::new(6), Ref::new(7), Ref::new(8)];
+    {
+        // this field doens't hold much of a value itself, it simply groups it's
+        // child annotations together so the reader knows they are connected
+        // without this, we'd simply have checkboxes
+        let mut field = pdf.form_field(radio_id);
+        field
+            .partial_name(TextStr("radio"))
+            .field_type(FieldType::Button)
+            .field_flags(
+                FieldFlags::RADIO
+                    | FieldFlags::NO_TOGGLE_TO_OFF
+                    | FieldFlags::RADIOS_IN_UNISON,
+            )
+            .children(radio_ids);
+    }
+
+    // we place all three buttons side by side so the user can tell they are
+    // one set of radio buttons, but in theory they can be anywhere even on
+    // different pages, two of those will have the same appearance, this means
+    // they will be turned on and off in unison because we used RADIOS_IN_UNISON
+    let radios = [
+        (Rect::new(108.0, 710.0, 138.0, 728.0), b"ch1"),
+        (Rect::new(140.0, 710.0, 170.0, 728.0), b"ch2"),
+        (Rect::new(172.0, 710.0, 202.0, 728.0), b"ch1"),
+    ];
+
+    let radio_on_appearance_id = Ref::new(9);
+    let radio_off_appearance_id = Ref::new(10);
+
+    // each actual button now get's its own annotation
+    for (id, (rect, state)) in radio_ids.into_iter().zip(radios) {
+        let mut field = pdf.form_field(id);
+        field.parent(radio_id);
+
+        let mut annot = field.to_annotation();
+        annot.rect(rect);
+
+        // we state which appearance state each button has by default
+        annot.appearance_state(Name(b"Off"));
+
+        // most readers need appearances to distinguish the actual radio buttons
+        // the normal appearance should contain an appearance stream for each
+        // state of the button, each button should have a unique name for it's
+        // on state and it's `/Off` state, if the on-state names are not unique
+        // they will shate their state, we see this here with ch1 being in there
+        // twice
```

Comment:
These comments would be much easier to read with some full stops in them. I would personally write all comments as complete, capitalized sentences ended with a full stop. This way, a short comment can be extended to a multi-sentence comment without it getting inconsistent.